### DISCUSSION TITLE
WIKI Updater: repository does not exist or you do not have permissions

### DISCUSSION THREAD

**Discussion by COWI-FIBR** [rfennell/AzurePipelines#1191] [Q&A]
I've read through several of the Q&A and issues related to authentication. Unfortunately, none of these seem to solve ​the issue for me.

I'm trying to update a [single wiki page](https://github.com/rfennell/AzurePipelines/wiki/WIKI-Updater-Tasks#update-a-single-wiki-page) in [DevOps](https://github.com/rfennell/AzurePipelines/wiki/WIKI-Updater-Tasks#azure-devops-services--azure-devops-server-tfs-wikis).

I've tried both [OAUTH](https://github.com/rfennell/AzurePipelines/wiki/WIKI-Updater-Tasks#authentication-using-oauth-to-azure-devops-services-hosted-repos) and [PAT](https://github.com/rfennell/AzurePipelines/wiki/WIKI-Updater-Tasks#authentication-using-personal-access-tokens-to-github-hosted-repos). The error message is the same regardless of authentication method.

My DevOps user is a member of the Project administrators group. 

From task log with system debug:
`##[debug]Processed: ##vso[task.issue type=error;]Error: Cloning into '/home/vsts/work/1/s\repo'...%0Aremote: TF401019: The Git repository with name or identifier ReleaesNoteTest.wiki does not exist or you do not have permissions for the operation you are attempting.%0Afatal: repository 'https://cowigroup.visualstudio.com/Agile%2520Test/_git/ReleaesNoteTest.wiki/' not found%0A`

NB. Yes, there's a spelling error in the project name "**Releaes**NoteTest".

Task with OAUTH
> task: WikiUpdaterTask@1
  inputs:
    repo: 'https://XXX.visualstudio.com/YYY/_git/ReleaesNoteTest.wiki'
    filename: 'Release notes.md'
    replaceFile: true
    dataIsFile: true
    sourceFile: 'releasenotes.md'
    message: 'Added release note to wiki'
    gitname: '$(Build.RequestedForId)'
    gitemail: '$(Build.RequestedForEmail)'
    useAgentToken: true
    localpath: '$(System.DefaultWorkingDirectory)\repo'
    fixSpaces: true
    insertLinefeed: true

Task with PAT
> task: WikiUpdaterTask@1
  inputs:
    repo: 'https://XXX.visualstudio.com/YYY/_git/ReleaesNoteTest.wiki'
    filename: 'Release notes.md'
    replaceFile: false
    appendToFile: true
    dataIsFile: true
    sourceFile: 'releasenotes.md'
    message: 'Added release note to wiki'
    gitname: '$(Build.RequestedForId)'
    gitemail: '$(Build.RequestedForEmail)'
    useAgentToken: false
    user: 'XYZ'
    password: '$(PAT)'
    localpath: '$(System.DefaultWorkingDirectory)\repo'
    fixSpaces: true
    insertLinefeed: true

Any help is much appreciated.

**Comment by rfennell** [comment]
This is usually one of two issues
-  The URL is wrong
- You don't have permissions

First I would check the URL
1. Can you use that URL in a browser and navigate to the repo?
2. If this an Azure DevOps project wiki or a publish as code one? If the formed make sure you get the URL from the wiki menu

![image](https://user-images.githubusercontent.com/324718/149781568-21423ada-c856-43bc-b5c4-6518b5c31691.png)

You need to do this as the format is a bit different to standard code publish repo URLs

After that is nearly aways down to the user the task is running as is missing permissions

**Comment by COWI-FIBR** [comment]
Hi rfennell,

Thanks for the quick response.

1. Yes, removing `.wiki` at the end I can navigate to the repo in a browser.
2.  I'm guessing the latter since there's no `Clone wiki` option in the dropdown.
![image](https://user-images.githubusercontent.com/57252942/149786291-92127525-82c7-4d43-9953-01fa664fd061.png)
I'm using the address that comes from cloning the repository.
![clone](https://user-images.githubusercontent.com/57252942/149785345-e40868a2-8015-4d37-aa35-396314c9b4df.png)

**Reply by rfennell** [reply]
Ah, as there is no clone option on the menu it means the WIKI is basically a standard repo that has been published (or folders within it published) as a WIKI. This means you can browse to it in the standard repo's area, see the contents as raw files, get the clone URL and set security. 

![image](https://user-images.githubusercontent.com/324718/149798091-75abc668-4816-42e8-bd99-e88a020a640b.png)

![image](https://user-images.githubusercontent.com/324718/149798138-f046cd13-7d1f-4209-ba3e-a73153843b28.png)


You only see the clone on the wiki menu the WIKI is a 'project wiki' i.e. the type created by default if you have no WIKI and browse to the WIKI page. These project WIKI repo's don't appear in the repo list - confusing or what !

So in your case make sure the URL is correct from the repos menu, and the permission are set on the correct repo

**Reply by COWI-FIBR** [reply]
Thank you for your response.

This was my bad. There is a Clone option for the wiki, I was just looking in the wrong place. (correct place is Wiki repo, I have no idea how I can overlooked this...)

Issue with authentication via OAUTH persist, but now that I've got the correct cloned repo URL I'm using PAT as this is easiest authentication option.
The end goal is to export the wiki to pdf. Once that is achieved I'll return to figure out the OAUTH issue.

**Reply by rfennell** [reply]
The OAUTH issue will be down to one of the following

**The User**
The correct build service user does not have the contribute permission set for the repo. Thus will be either the `project Collection Build Service` or `<project name> Build Service`.

Which one is based on the `Project Settings > Pipelines > Settings > Limit job authorization scope to current project for non-release pipelines`

![image](https://user-images.githubusercontent.com/324718/151187118-1352ff2f-f41e-4c85-8712-518d61344f59.png)

Repo Projects
Also make sure that the `Project Settings >  Pipeline > Setting > Limit job authorization scope to referenced Azure DevOps repositories', or as I think it is now called `Protect access to repositories in YAML pipelines`. If this set correctly it can block access to the target repo.

**Reply by COWI-FIBR** [reply]
It was the **Protect access to repositories in YAML pipelines** that needed disabling, thanks.

**Reply by rfennell** [reply]
I should update the documentation as that label has changed, glad you are sorted
